pkg/cloudprovider/servicecontroller/servicecontroller.go

```diff
@@ -225,44 +225,25 @@ func (s *ServiceController) createLoadBalancerIfNeeded(namespacedName types.Name
 		glog.Infof("LB already exists and doesn't need update for service %s", namespacedName)
 		return nil, notRetryable
 	}
-	if cachedService != nil {
```

Comment:
Getting rid of this deletion here will surely break GCE at head, because GCE's CreateTCPLoadBalancer function doesn't currently handle updating the LB's parameters.